Kernel/Arch/x86/RegisterState.h

```diff
@@ -109,13 +109,21 @@ struct [[gnu::packed]] RegisterState {
         arg2 = ecx;
         arg3 = ebx;
         arg4 = esi;
-#else
+#elif ARCH(X86_64)
         // The syscall instruction clobbers rcx, so we must use a different calling convention to 32-bit.
         function = rax;
         arg1 = rdx;
         arg2 = rdi;
         arg3 = rbx;
         arg4 = rsi;
+#elif ARCH(AARCH64)
```

Comment:
This change should not be here. We should never be in Arch/x86/RegisterState.h if ARCH(AARCH64) is defined. this file has `VALIDATE_IS_X86()` at the top.

Reply:
hah! very true! 🫣